ledger/recovering-indexer-integration-tests/src/test/suite/scala/com/digitalasset/platform/indexer/RecoveringIndexerIntegrationSpec.scala

```diff
@@ -308,10 +312,15 @@ object RecoveringIndexerIntegrationSpec {
           committerExecutionContext = committerExecutionContext,
         )
       } yield new KeyValueParticipantState(
-        readerWriter,
-        readerWriter,
-        metrics,
-        enableSelfServiceErrorCodes = true,
+        KeyValueParticipantStateReader(
```

Comment:
By introducing a factory method for `KeyValueParticipantState` that creates the adaptors from a combined reader/writer, this could be greatly simplified and the construction logic moved to a single place.